Ensure static analysis respects go mod

Static analysis started failing on this upstream change. But should not have based on the go mod config.

```
Error: lxd/daemon.go:1814:17: cannot use idmClient (variable of type *candidclient.Client) as "gopkg.in/macaroon-bakery.v2/bakery/identchecker".IdentityClient value in struct literal: *candidclient.Client does not implement "gopkg.in/macaroon-bakery.v2/bakery/identchecker".IdentityClient (wrong type for method DeclaredIdentity)
		have DeclaredIdentity(ctx "context".Context, declared map[string]string) ("gopkg.in/macaroon-bakery.v3/bakery/identchecker".Identity, error)
		want DeclaredIdentity(ctx "context".Context, declared map[string]string) ("gopkg.in/macaroon-bakery.v2/bakery/identchecker".Identity, error) (typecheck)
		client:       idmClient,
		              ^
make: *** [Makefile:250: static-analysis] Error 1
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -31,9 +31,9 @@ jobs:
         run: |
           make deps
 
-      - name: Update gomod
+      - name: Download go dependencies
         run: |
-          make update
+          go mod download
 
       - name: Run static analysis
         run: |

